### PR TITLE
Auth changes for meeting-based login

### PR DESCRIPTION
**Pull request by Baggerone**
Does not allow for social auth options yet.
At Billy's suggestion, I included the idea of an InviteType, in order to allow for future invites that are for things that aren't meetings.

### CODE REVIEW

File: application/models/post_test.go [hunk 1]
```diff
@@ -1690,16 +1690,17 @@ func (ms *ModelSuite) TestPosts_FindByUser() {
 func (ms *ModelSuite) TestPosts_GetPotentialProviders() {
 	t := ms.T()
 
-	f := createFixturesFor_Posts_GetPotentialProviders(ms)
```
Comment (Schparky) [inline application/models/post_test.go]: can `createFixturesFor_Posts_GetPotentialProviders` be removed now?

File: application/actions/auth.go [hunk 2]
```diff
@@ -30,6 +30,14 @@ const (
...
 	ExpiresUTCParam = "expires-utc"
 
+	// http params for the Invite type and code
+	InviteCode        = "code"
+	InviteTypeMeeting = "meeting"
+
+	// session keys for using invites for authentication
+	InviteTypeSessionKey       = "InviteType"
+	InviteObjectUUIDSessionKey = "InviteObjectID"
+
```
Comment (Schparky) [inline application/actions/auth.go]: Maybe prefix the http param names with something like `QueryParam` and the sessions keys with `SessionKey`. (The session keys have it at the end, which seems a bit different than our norm.)

Actually, I just looked at the surrounding code and the "norm" for this file is to have the "category" at the end, so maybe just add `Param` to the first two, unless you feel like changing all of them.

Comment (Baggerone) [reply application/actions/auth.go]: :+1:

File: application/gqlgen/schema.graphql [hunk 3]
```diff
@@ -112,8 +107,8 @@ type Mutation {
     NOT YET IMPLEMENTED --
     Validate a `MeetingInvite` and create a new `MeetingParticipant`. If the `MeetingParticipant.User` contains
     one or more `Organizations`, the user should be redirected to the standard login page. Otherwise, login should
-    proceed through one of the configured `AuthType`s described by the `systemConfig` query.
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/gqlgen/schema.graphql]: OK.

File: application/models/post_test.go (hunk 1 above)
Comment (Baggerone) [reply application/models/post_test.go]: :+1:

File: application/models/meetingparticipant.go [hunk 4]
```diff
@@ -55,3 +55,14 @@ func (m *MeetingParticipant) Invite() (*MeetingInvite, error) {
 	}
 	return &invite, nil
 }
+
+func (m *MeetingParticipant) FindByMeetingIDAndUserID(meetingID, userID int) error {
+	return DB.Where("meeting_id = ? and user_id = ?", meetingID, userID).First(m)
+}
+
+func (m *MeetingParticipant) CreateForInvite(invite MeetingInvite, userID int) error {
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/models/meetingparticipant.go]: OK

**Pull request merged**